rp2040-hal/src/multicore.rs

```diff
@@ -170,10 +176,19 @@ impl<'p> Core<'p> {
                 let mut sio = Sio::new(peripherals.SIO);
                 sio.fifo.write_blocking(1);
 
-                entry()
+                entry();
+                // TODO: should this bring the core back to its startup state waiting for a
+                // function? and/or reset its NVIC and other core specific blocks.
+                loop {
+                    cortex_m::asm::wfe()
+                }
             }
 
             // Reset the core
+            // TODO: resetting without prior check that the core is actually stowed is unsafe.
```

Comment:
Is it unsafe in the sense that it breaks any of rust's guarantees? Or just a bad idea, for example because it might cause deadlocks if the other core is inside a critical section when it gets reset?

Reply:
I think it is unsafe because it can cause cases "poisoning" (of data and peripherals) without a way to detect it.
As far as I'm aware, threads in std cannot be killed nor recycle so I'm not sure about the entire scope of the consequences but I definitely "feels wrong" to just kill it.

That being said, I am not aware of a way to tell if core1 is stowed so this would require some bookkeeping in the hal to:
- avoid a new spawn while the first one hasn't gracefully shut down
- capture panics and clear/mark the book to allow graceful recovery.

I'm not even sure the latter can be done.

Reply:
Isn't resetting the core very similar to a theoretical situation where the core gets stuck in some infinite loop, and a new thread gets started on another core? I don't see how this is unsafe. Of course it can cause deadlocks in case the core had locks on some resources which now wouldn't get released. But again, that's not different from an infinite loop inside a critical section. Definitely not a good idea, but still not unsafe.

Reply:
I see what you mean. My choice of word isn't great there indeed.
It is not `unsafe` in the rust sense, but it is a `bad` thing to do.

I'm removing this comment to avoid confusion.

Reply:
After further thinking about it, I must admit that reasoning about the implications of such a reset is really tricky. 

While I was not yet able to construct something which would trigger obviously undefined behavior, I'm far from confident that it's really sound.

Reply:
I think this discussion is not related to the merge request itself. We don't introduce a regression. The safety of resetting a running core can be discussed separately.